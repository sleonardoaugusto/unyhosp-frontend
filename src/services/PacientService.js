import apiClient from './config'

export default {
  getPacients() {
    return apiClient.get('/pacients/')
  },

  postPacient(data) {
    return apiClient.post('/pacients/', data)
  }
}