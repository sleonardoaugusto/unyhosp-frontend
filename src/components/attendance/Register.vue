<template>
  <div class="d-flex flex-wrap justify-content-center align-items-center flex-column">
    <div class="col-md-6 mb-3">
      <div class="row">
        <h2 class="type-h4 type-light-green">Registrar atendimento</h2>
      </div>
    </div>
    <div class="col-md-6">
      <div class="row d-flex flex-column mb-3">
        <label for="PacientName">Paciente:</label>
        <input type="text" id="PacientName" v-model="inputValue" @input="searchPacients">
        <ul class="live-search__dropdown">
          <li class="live-search__item" v-for="(p, i) in pacients" :key="i" @click="setPacient(p)">{{ p.name }}</li>
        </ul>
      </div>
    </div>
  </div>
</template>

<script>
import PacientService from '@/services/PacientService'

export default {
  props: ['hospitalId'],
  name: 'Register',
  data: () => ({
    inputValue: '',
    pacients: []
  }),
  methods: {
    async searchPacients() {
      await PacientService.getPacients()
        .then((res) => {
          this.pacients = res.data
        })
    },
    setPacient(value) {
      this.inputValue = value.name
    }
  }
}
</script>

<style scoped>

</style>