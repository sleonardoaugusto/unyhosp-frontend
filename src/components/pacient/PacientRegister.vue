<template>
  <div class="d-flex flex-wrap justify-content-center align-items-center flex-column">
    <div class="col-md-6 mb-3">
      <h2 class="type-h4 type-blue-smoke">Cadastrar Paciente</h2>
    </div>
    <form @submit.prevent="submit" class="col-md-6">
      <div class="form-group" :class="{ 'form-group--error': $v.pacient.name.$error }">
        <label class="form__label" for="PacientName">Nome:</label>
        <input class="form__input" type="text" id="PacientName" v-model.trim.lazy="$v.pacient.name.$model">
        <div v-if="!$v.pacient.name.required" class="error--message">Campo obrigatório</div>
      </div>
      <div class="form-group" :class="{ 'form-group--error': $v.pacient.document_id.$error }">
        <label class="form__label" for="DocumentId">CPF:</label>
        <the-mask class="form__input" type="text" id="DocumentId" mask="###.###.###-##"
                  v-model.trim.lazy="$v.pacient.document_id.$model"/>
        <div v-if="!$v.pacient.document_id.required" class="error--message">Campo obrigatório</div>
        <div v-if="!$v.pacient.document_id.minLength" class="error--message">CPF inválido</div>
      </div>
      <div class="form-group" :class="{ 'form-group--error': $v.pacient.email.$error }">
        <label class="form__label" for="Email">Email:</label>
        <input class="form__input" type="text" id="Email" v-model.trim.lazy="$v.pacient.email.$model">
        <div v-if="!$v.pacient.email.required" class="error--message">Campo obrigatório</div>
        <div v-if="!$v.pacient.email.email" class="error--message">Email inválido</div>
      </div>
      <div class="form-group" :class="{ 'form-group--error': $v.pacient.date_of_birth.$error }">
        <label class="form__label" for="DateOfBirth">Data de Nascimento:</label>
        <input class="form__input" type="text" id="DateOfBirth" v-mask="'##/##/####'"
               v-model.trim.lazy="$v.pacient.date_of_birth.$model">
        <div v-if="!$v.pacient.date_of_birth.required" class="error--message date">Campo obrigatório</div>
        <div v-if="!$v.pacient.date_of_birth.minLength" class="error--message date">Data inválida</div>
      </div>
      <button class="button button-grey" type="submit">CADASTRAR</button>
    </form>
  </div>
</template>

<script>
import PacientService from '@/services/PacientService'
import { required, email, minLength } from 'vuelidate/lib/validators'

export default {
  name: 'PacientRegister',
  data: () => ({
    pacient: {
      name: '',
      document_id: '',
      email: '',
      date_of_birth: ''
    }
  }),
  validations: {
    pacient: {
      name: { required },
      email: { required, email },
      document_id: { required, minLength: minLength(11) },
      date_of_birth: { required, minLength: minLength(10) }
    }
  },
  methods: {
    submit() {
      this.$v.$touch()
      if (!this.$v.$invalid) {
        PacientService.post(this.pacient)
      }
    }
  }
}
</script>

<style scoped>

</style>