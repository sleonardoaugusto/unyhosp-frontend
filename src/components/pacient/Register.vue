<template>
  <div class="d-flex flex-wrap justify-content-center align-items-center flex-column">
    <div class="col-md-6 mb-3">
      <h2 class="type-h4 type-blue-smoke">Cadastrar Paciente</h2>
    </div>
    <form @submit.prevent="submit" class="col-md-6">
      <div class="form-group" :class="{ 'form-group--error': $v.data.name.$error }">
        <label class="form__label" for="PacientName">Nome:</label>
        <input class="form__input" type="text" id="PacientName" v-model.trim.lazy="$v.data.name.$model">
        <div v-if="!$v.data.name.required" class="error--message">Campo obrigatório</div>
      </div>
      <div class="form-group" :class="{ 'form-group--error': $v.data.document_id.$error }">
        <label class="form__label" for="DocumentId">CPF:</label>
        <the-mask class="form__input" type="text" id="DocumentId" mask="###.###.###-##"
                  v-model.trim.lazy="$v.data.document_id.$model"/>
        <div v-if="!$v.data.document_id.required" class="error--message">Campo obrigatório</div>
        <div v-if="!$v.data.document_id.minLength" class="error--message">CPF inválido</div>
      </div>
      <div class="form-group" :class="{ 'form-group--error': $v.data.email.$error }">
        <label class="form__label" for="Email">Email:</label>
        <input class="form__input" type="text" id="Email" v-model.trim.lazy="$v.data.email.$model">
        <div v-if="!$v.data.email.required" class="error--message">Campo obrigatório</div>
        <div v-if="!$v.data.email.email" class="error--message">Email inválido</div>
      </div>
      <div class="form-group" :class="{ 'form-group--error': $v.data.date_of_birth.$error }">
        <label class="form__label" for="DateOfBirth">Data de Nascimento:</label>
        <input class="form__input" type="tel" id="DateOfBirth" v-mask="'##/##/####'"
               v-model.trim.lazy="$v.data.date_of_birth.$model">
        <div v-if="!$v.data.date_of_birth.required" class="error--message">Campo obrigatório</div>
      </div>
      <button class="button button-grey" type="submit">CADASTRAR</button>
    </form>
  </div>
</template>

<script>
import PacientService from '@/services/PacientService'
import { required, email, minLength } from 'vuelidate/lib/validators'

export default {
  name: 'Register',
  data: () => ({
    data: {
      name: '',
      document_id: '',
      email: '',
      date_of_birth: ''
    },
    submitStatus: null
  }),
  validations: {
    data: {
      name: { required },
      email: { required, email },
      document_id: { required, minLength: minLength(11) },
      date_of_birth: { required }
    }
  },
  methods: {
    submit() {
      console.log('submit!')
      this.$v.$touch()
      if (this.$v.$invalid) {
        this.submitStatus = 'ERROR'
      } else {
        // do your submit logic here
        this.submitStatus = 'PENDING'
        setTimeout(() => {
          this.submitStatus = 'OK'
          PacientService.post(this.data)
        }, 500)
      }
    }
  }
}
</script>

<style scoped>

</style>