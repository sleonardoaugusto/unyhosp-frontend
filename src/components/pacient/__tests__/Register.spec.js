import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import Register from '@/components/pacient/Register'
import VueTheMask from 'vue-the-mask'
import Vuelidate from 'vuelidate'
import PacientService from '@/services/PacientService'
import pacient from '@/components/pacient/__tests__/__mocks__/pacient'
import Vuex from 'vuex'
import flushPromises from 'flush-promises'

jest.mock('@/services/PacientService', () => ({
  post: () => (Promise.resolve())
}))

const localVue = createLocalVue()

localVue.use(VueRouter)
localVue.use(VueTheMask)
localVue.use(Vuelidate)
localVue.use(Vuex)

describe('Register.vue', () => {
  let wrapper

  let store
  let state
  let actions

  beforeEach(() => {
    state = {
      notifications: []
    }

    actions = {
      'notification/add': jest.fn()
    }

    store = new Vuex.Store({
      modules: {
        notification: {
          state,
          actions
        }
      }
    })

    wrapper = shallowMount(Register, {
      localVue,
      store
    })
  })

  it('component is rendering properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('feedback form validation should appear', () => {
    wrapper.setData({ pacient: pacient.invalidData })
    wrapper.vm.$forceUpdate()
    const errors = wrapper.findAll('.error--message')
    expect(errors.at(0).text()).toBe('Campo obrigat칩rio')
    expect(errors.at(1).text()).toBe('CPF inv치lido')
    expect(errors.at(2).text()).toBe('Email inv치lido')
    expect(errors.at(3).text()).toBe('Data inv치lida')
  })

  it('feedback form validation should NOT appear', () => {
    wrapper.setData({ pacient: pacient.validData })
    wrapper.vm.$forceUpdate()
    expect(wrapper.findAll('.error--message').length).toBe(0)
  })

  it('click button should call submit method', () => {
    const submit = jest.fn()
    wrapper.setMethods({ submit })
    wrapper.find('form').trigger('submit')
    expect(submit).toHaveBeenCalled()
  })

  it('PacientService post method should NOT be called on submit if data is INVALID', () => {
    wrapper.setData({ pacient: pacient.invalidData })
    const serviceSpy = jest.spyOn(PacientService, 'post')
    wrapper.find('form').trigger('submit')
    expect(serviceSpy).not.toHaveBeenCalled()
  })

  it('PacientService post method SHOULD be called on submit if data is VALID', () => {
    wrapper.setData({ pacient: pacient.validData })
    const serviceSpy = jest.spyOn(PacientService, 'post')
    wrapper.find('form').trigger('submit')
    expect(serviceSpy).toHaveBeenCalledWith(pacient.validData)
  })

  it('Promise resolved should call notification add store module method', async () => {
    wrapper.setData({ pacient: pacient.validData })
    wrapper.find('form').trigger('submit')
    await flushPromises()
    expect(actions['notification/add']).toHaveBeenCalled()
  })
})