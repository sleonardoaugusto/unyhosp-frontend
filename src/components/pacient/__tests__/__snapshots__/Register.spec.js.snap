// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pacient/Register.vue component is rendering properly 1`] = `
<div class="d-flex flex-wrap justify-content-center align-items-center flex-column">
  <div class="col-md-6 mb-3">
    <h2 class="type-h4 type-blue-smoke">Cadastrar Paciente</h2>
  </div>
  <form class="col-md-6">
    <div class="form-group"><label for="PacientName" class="form__label">Nome:</label> <input type="text" id="PacientName" class="form__input">
      <div class="error--message">Campo obrigat贸rio</div>
    </div>
    <div class="form-group"><label for="DocumentId" class="form__label">CPF:</label>
      <the-mask-stub value="" mask="###.###.###-##" tokens="[object Object]" type="text" id="DocumentId" class="form__input"></the-mask-stub>
      <div class="error--message">Campo obrigat贸rio</div>
      <!---->
    </div>
    <div class="form-group"><label for="Email" class="form__label">Email:</label> <input type="text" id="Email" class="form__input">
      <div class="error--message">Campo obrigat贸rio</div>
      <!---->
    </div>
    <div class="form-group"><label for="DateOfBirth" class="form__label">Data de Nascimento:</label> <input type="tel" id="DateOfBirth" class="form__input">
      <div class="error--message">Campo obrigat贸rio</div>
    </div> <button type="submit" class="button button-grey">CADASTRAR</button>
  </form>
</div>
`;
